packages/phone-number-privacy/combiner/src/common/crypto-clients/crypto-client.ts

```diff
@@ -0,0 +1,57 @@
+import { ErrorMessage, KeyVersionInfo } from '@celo/phone-number-privacy-common'
+import Logger from 'bunyan'
+
+export interface ServicePartialSignature {
+  url: string
+  signature: string
+}
+
+export abstract class CryptoClient {
+  protected unverifiedSignatures: ServicePartialSignature[] = []
+
+  constructor(protected readonly keyVersionInfo: KeyVersionInfo) {}
+
+  /**
+   * Returns true if the number of valid signatures is enough to perform a combination
+   */
+  public hasSufficientSignatures(): boolean {
+    return this.allSignaturesLength >= this.keyVersionInfo.threshold
+  }
+
+  public addSignature(serviceResponse: ServicePartialSignature): void {
+    this.unverifiedSignatures.push(serviceResponse)
+  }
+
+  /*
+   * Computes the signature for the blinded phone number using subclass-specific
+   * logic defined in _combinePartialBlindedSignatures.
+   * Throws an exception if not enough valid signatures.
+   */
+  public combinePartialBlindedSignatures(blindedMessage: string, logger: Logger): string {
```

Comment:
```suggestion
  public combineBlindedSignatureShares(blindedMessage: string, logger: Logger): string {
```
You might clarify that this is a partial blinded-signature and not a partially-blinded signature. Here is one suggestion.

Reply:
I'll also note that this abstraction hides a sharp edge, which is that although the BLS implementation is verifying, the POPRF implementation is not. In a cryptography library design, this is not ideal since it increases risk for misunderstanding therefore bugs. I think this risk is relatively minor in this case and so am not necessarily suggesting we refactor this now.

Reply:
@eelanagaraj - changed name in my PR fixes branch